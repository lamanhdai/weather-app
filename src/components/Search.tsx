import React, { useRef} from 'react';
import {
  Form,
  FormControl
} from 'react-bootstrap';

interface SearchProps {
  onSearch: (keyword: string) => void
}

function Search(props:SearchProps) {
  const searchRef = useRef<HTMLInputElement>(null);
  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    const form = event.currentTarget;
    event.preventDefault();
    if (form.checkValidity() === false) {
      return;
    }
    if(searchRef.current) {
      props.onSearch(searchRef.current.value)
    }
  };
  return (
    <div className="mt-5">
      <Form onSubmit={handleSubmit}>
        <Form.Group className="form-group">
          <FormControl type="text" aria-describedby="search" role="search" placeholder="Search a location" ref={searchRef} required />
        </Form.Group>
      </Form>
    </div>
  );
}

export default Search;
