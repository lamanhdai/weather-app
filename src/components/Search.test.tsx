import React from 'react';
import { mount, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from './Search';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
const contextValue = {
  dispatch: jest.fn(),
};

describe('Search', () => {
  it('search element', () => {
    const SearchElement = renderer
      .create(<Search onSearch={contextValue.dispatch} />)
      .toJSON();
    expect(SearchElement).toMatchSnapshot();
  });

  it('render', () => {
    const wrapper = mount(
      <Search onSearch={contextValue.dispatch} />
    );
    expect(wrapper.find('input').props().role).toEqual('search');
  });
});