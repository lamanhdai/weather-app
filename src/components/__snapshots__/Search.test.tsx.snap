// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search search element 1`] = `
<div
  className="mt-5"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group form-group"
    >
      <input
        aria-describedby="search"
        className="form-control"
        placeholder="Search a location"
        required={true}
        role="search"
        type="text"
      />
    </div>
  </form>
</div>
`;
